nexusutils/dataconverter/readers/em_spctrscpy/utils/em_event_data.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+"""Representation of an NXevent_data_em class."""
+
+# -*- coding: utf-8 -*-
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=E1101
+
+
+# from nexusutils.dataconverter.readers.em_spctrscpy.utils.oina.em_oina_xray \
+#     import NxOinaSpectrumSetEmXray
+
+import datetime
+
+from typing import Dict
+
+import hyperspy.api as hs
+
+from nexusutils.dataconverter.readers.em_spctrscpy.utils.em_nexus_base_classes \
+    import NxObject
+
+from nexusutils.dataconverter.readers.em_spctrscpy.utils.hspy.em_hspy_xray \
+    import NxSpectrumSetEmXray
+
+from nexusutils.dataconverter.readers.em_spctrscpy.utils.hspy.em_hspy_eels \
+    import NxSpectrumSetEmEels
+
+from nexusutils.dataconverter.readers.em_spctrscpy.utils.hspy.em_hspy_adf \
+    import NxImageSetEmAdf
+
+
+class NxEventDataEm:
+    """Representing a data collection event with a single detector.
+
+    During this data collection event, the microscope
+    was considered stable enough.
+    """
+
+    def __init__(self, file_name: str, entry_id: int):
+        assert file_name != "", "Argument file_name must not be empty !"
+        assert entry_id > 0, "Argument entry_id has to be > 0 !"
```

Comment:
Raise exceptions instead of assert here.

Reply:
As mentioned in another comment, the suggestion is good, haven't experimented much with these, will do so in em_oim_ms and then proliferate these through to em_spctrscpy and apm